aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -1541,10 +1552,46 @@ TORCH_IMPL_FUNC(any_out)
   allany_impl<0>(self, result, dim, keepdim, or_stub);
 }
 
+TORCH_IMPL_FUNC(any_dims_out)
+(const Tensor& self, OptionalIntArrayRef dim, bool keepdim, const Tensor& result) {
+  allany_impl<0>(self, result, dim, keepdim, or_stub);
+}
+
 TORCH_IMPL_FUNC(any_all_out)(const Tensor& self, const Tensor& result) {
   allany_impl<0>(self, result, {}, false, or_stub);
 }
 
+template <bool is_all>
+Tensor allany_dims_default(const Tensor &self, OptionalIntArrayRef dim, bool keepdim) {
+  // Default implementation in terms of all-reduce or single dim reduce
+  if (!dim) {
+    auto out = is_all ? self.all() : self.any();
+    if (keepdim) {
+      DimVector out_shape(self.dim(), 1);
+      return out.expand(out_shape);
+    }
+    return out;
+  }
+
+  Tensor out = self;
+  for (auto d : *dim) {
```

Comment:
This is fine, but couldn't we extend the tensoriterator implementation? I reckon it should be straightforward and it should be much more efficient.

Reply:
CPU and CUDA do use tensoriterator. I included this mainly for MPS or any third-party dispatch keys.